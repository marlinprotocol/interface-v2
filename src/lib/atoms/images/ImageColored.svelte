<script lang="ts">
	import type { CommonVariant } from '$lib/types/componentTypes';
	import { getImageClassByVariant } from '$lib/utils/helpers/componentHelper';

	export let src: string;
	export let alt = 'Image';
	export let variant: CommonVariant = 'primary';
	export let width = 13;

	const styleClass = getImageClassByVariant(variant);
</script>

<img {src} {alt} class={styleClass} {width} />

<style>
	.primary-img {
		filter: invert(22%) sepia(80%) saturate(6978%) hue-rotate(229deg) brightness(75%) contrast(100%);
	}
	.success-img {
		filter: invert(47%) sepia(54%) saturate(2466%) hue-rotate(86deg) brightness(97%) contrast(116%);
	}
	.error-img {
		filter: invert(11%) sepia(100%) saturate(6417%) hue-rotate(11deg) brightness(86%) contrast(117%);
	}
	.secondary-img {
		filter: invert(0%) sepia(0%) saturate(0%) hue-rotate(202deg) brightness(100%) contrast(104%);
	}
	.warning-img {
		filter: invert(72%) sepia(22%) saturate(6859%) hue-rotate(359deg) brightness(103%)
			contrast(105%);
	}
	.disabled-img {
		filter: invert(54%) sepia(80%) saturate(6978%) hue-rotate(229deg) brightness(89%) contrast(84%);
	}
	.grey-img {
		filter: invert(50%) sepia(10%) saturate(10%) hue-rotate(11deg) brightness(100%) contrast(104%);
	}
</style>
