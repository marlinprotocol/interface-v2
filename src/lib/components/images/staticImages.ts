const Alert = '/images/alert.svg';
const Arrow = '/images/arrowicon.svg';
const ArrowLong = '/images/arrow-long.svg';
const ArrowRightSolid = '/images/arrowRightSolid.svg';
const ArrowLeftSolid = '/images/arrowLeftSolid.svg';
const BlueAlert = '/images/blue-alert.svg';
const Check = '/images/vectorcheck.svg';
const CloseCircle = '/images/close-circle-icon.svg';
const Copy = '/images/copyicon.svg';
const CopyGrey = '/images/copy-btn.svg';
const Cycle = '/images/cycleimg.svg';
const CycleLoader = '/images/cycleloader.svg';
const History = '/images/historyicon.svg';
const Info = '/images/info.svg';
const LeftBlueArrow = '/images/left-blue-arrow.svg';
const Lock = '/images/lockicon.svg';
const Metamask = '/images/metamaskicon.svg';
const OpenInNew = '/images/openinnew.svg';
const RightArrow = '/images/rightarrow.svg';
const RightArrowWhite = '/images/rightarrowwhite.svg';
const Sheild = '/images/shield.svg';
const Sort = '/images/sort.svg';
const Timer = '/images/timerclock.svg';
const UnActiveTick = '/images/unactivetick.svg';
const Verified = '/images/verifiedicon.svg';
const WalletConnect = '/images/walleticon.svg';
const WalletConnected = '/images/wallet-connected.svg';
const PolygonLogo = '/images/polygon.svg';
const PolygonFullLogo = '/images/polygon-full.webp';
const ArbitrumLogo = '/images/arbitrum.webp';
const EthereumLogo = '/images/eth.webp';
const LineaLogo = '/images/linea.webp';
const MarlinLogo = 'https://blog.marlin.org/wp-content/uploads/2024/01/200x200.png';
const navButton = '/images/navbtn.svg';
const chainLogo = '/images/chainSwitchLogo.svg';
const discChainLog = '/images/discChain.svg';
const bridgeIcon = '/images/bridgeIcon.svg';
const bridgeIconBlue = '/images/bridgeIconBlue.svg';
const dashboardIcon = '/images/dashboardIcon.svg';
const dashboardIconBlue = '/images/dashboardIconBlue.svg';
const ecosystemIcon = '/images/ecosystemIcon.svg';
const ecosystemIconBlue = '/images/ecosystemIconBlue.svg';
const faqsIcon = '/images/faqsIcon.svg';
const faqsIconBlue = '/images/faqsIconBlue.svg';
const governanceIcon = '/images/governanceIcon.svg';
const governanceIconBlue = '/images/governanceIconBlue.svg';
const kalypsoIcon = '/images/kalypsoIcon.svg';
const kalypsoIconBlue = '/images/kalypsoIconBlue.svg';
const oysterIcon = '/images/oysterIcon.svg';
const oysterIconBlue = '/images/oysterIconBlue.svg';
const relayIcon = '/images/relayIcon.svg';
const relayIconBlue = '/images/relayIconBlue.svg';
const dataTransferIcon = '/images/dataTransferIcon.svg';
const dataTransferIconBlue = '/images/dataTransferIconBlue.svg';
const marlinSmLogo = '/images/marlinSmLogo.svg';
const marlinLgLogo = '/images/marlinLgLogo.svg';
const alertV2Icon = '/images/alertV2Icon.svg';
const sortV2Icon = '/images/sortV2Icon.svg';
const lightModeLogo = '/images/lightModeLogo.svg';
const developerChatIcon = '/images/developerChatIcon.svg';
const earthIcon = '/images/earth.svg';
const blogIcon = '/images/blog.svg';
const chevronDownIcon = '/images/chevronDownIcon.svg';
const arrowConversion = '/images/arrowConversion.svg';
const closeIcon = '/images/closeIcon.svg';
const successIcon = '/images/SuccessIcon.svg';
const alertRed = '/images/alertRed.svg';
const backIcon = '/images/backV2Icon.svg';

// history pages
const exchangeIcon = '/images/exchangeV2Icon.svg';
const cancelIcon = '/images/cancelV2Icon.svg';
const historyIcon = '/images/historyV2Icon.svg';
const conversionCycleIcon = '/images/conversionCycleV2Icon.svg';
const toastSuccessAlert = '/images/toastSuccessAlert.svg';
const toastErrorAlert = '/images/toastErrorAlert.svg';
const toastWarningAlert = '/images/toastWarningAlert.svg';
const conversionIcon = '/images/conversionIcon.svg';
const walleticonOyster = '/images/walletIconOyster.svg';
const clockIcon = '/images/clockIcon.svg';
const checkmark = '/images/checkmark.svg';
const tick = '/images/tick.svg';

const externalLinkIcon = '/images/externalLinkV2Icon.svg';
const searchV2Icon = '/images/searchV2Icon.svg';
const refreshV2Icon = '/images/refreshV2Icon.svg';
const infoV2Icon = '/images/infoV2Icon.svg';
const ankrLogo = '/images/ankr-blue-logo.webp';
const cradlesLogo = '/images/CradlesLogo.webp';
const brightlyStakeLogo = '/images/Brightlystakelogo.webp';
const midaSwapLogo = '/images/MidaswapLogo.webp';
const aspectaLogo = '/images/AspectaLogo.webp';
const echolinkLogo = '/images/EcholinkLogo.webp';
const publicAiLogo = '/images/PublicAIDarkLogo.webp';
const operatorLogo = '/images/operator.svg';
const sideLogo = '/images/side.svg';
const injectiveLogo = '/images/injective.webp';
const mellowLogo = '/images/mellow.webp';
const FlashBotsLogo = '/images/FlashBotsLogo.webp';
const gearBoxLogo = '/images/gearbox.webp';
const decimalLogo = '/images/decimal.webp';
const zkbobLogo = '/images/zkbob.svg';
const p2pLogo = '/images/p2p_logo_white.webp';
const t2t2Logo = '/images/T2T2Logo.png';
const hashkeyCloudLogo = '/images/HashkeyCloudLogo.webp';
const nodeasyLogo = '/images/NodeEasyLogo.webp';
const chorusOneLogo = '/images/ChorusOneLogo.webp';
const availLogo = '/images/availLogo.webp';
const stafiLogo = '/images/StafiLogo.webp';
const sjtUniversityLogo = '/images/sjtUni.webp';
const goSleepLogo = '/images/GoSleepLogo.webp';
const infstones = '/images/InfstonesLogo.webp';
const staking4allLogo = '/images/Staking4AllLogo.webp';
const degoLogo = '/images/DegoFinance.webp';
const promLogo = '/images/PromLogo.png';
const calendarIcon = '/images/calendar.svg';
const checkmarkCircleIcon = '/images/checkmark-circle.svg';
const clock = '/images/clock.svg';
const loadingIcon = '/images/loading.svg';
const yellowInfo = '/images/information-circle.svg';
const redAlert = '/images/alert-red.svg';
const greenTick = '/images/checkmark-circle-green.svg';
const deployIcon = '/images/deployIcon.svg';
const arrowUpRight = '/images/arrow-up-right.svg';
const backendServer = '/images/backend-servers.svg';
const chains = '/images/chains.svg';
const frontends = '/images/frontends.svg';
const instances = '/images/instances.svg';
const llms = '/images/llms.svg';
const regions = '/images/regions.svg';
const dnsLogo = '/images/3dns.svg';
const bonfidaLogo = '/images/bonfida.svg';
const spaceIdLogo = '/images/spaceid.svg';
const exploreIcon = '/images/explore.svg';
const copyIcon = '/images/copyIconNew.svg';
const infoIconPurple = '/images/information-purple.svg';

export const staticImages = {
	Alert,
	Arrow,
	ArrowLong,
	ArrowRightSolid,
	ArrowLeftSolid,
	BlueAlert,
	Check,
	CloseCircle,
	Copy,
	CopyGrey,
	Cycle,
	CycleLoader,
	History,
	Info,
	LeftBlueArrow,
	Lock,
	Metamask,
	OpenInNew,
	RightArrow,
	RightArrowWhite,
	Sheild,
	Sort,
	Timer,
	UnActiveTick,
	Verified,
	WalletConnect,
	WalletConnected,
	PolygonLogo,
	PolygonFullLogo,
	ArbitrumLogo,
	EthereumLogo,
	LineaLogo,
	MarlinLogo,
	navButton,
	chainLogo,
	discChainLog,
	bridgeIcon,
	bridgeIconBlue,
	dashboardIcon,
	dashboardIconBlue,
	dataTransferIcon,
	dataTransferIconBlue,
	ecosystemIcon,
	ecosystemIconBlue,
	faqsIcon,
	faqsIconBlue,
	governanceIcon,
	governanceIconBlue,
	kalypsoIcon,
	kalypsoIconBlue,
	oysterIcon,
	oysterIconBlue,
	relayIcon,
	relayIconBlue,
	marlinSmLogo,
	marlinLgLogo,
	alertV2Icon,
	sortV2Icon,
	lightModeLogo,
	developerChatIcon,
	earthIcon,
	blogIcon,
	chevronDownIcon,
	closeIcon,
	successIcon,
	alertRed,
	backIcon,
	exchangeIcon,
	cancelIcon,
	historyIcon,
	conversionCycleIcon,
	toastSuccessAlert,
	toastErrorAlert,
	toastWarningAlert,
	arrowConversion,
	conversionIcon,
	walleticonOyster,
	clockIcon,
	externalLinkIcon,
	checkmark,
	tick,
	searchV2Icon,
	refreshV2Icon,
	infoV2Icon,
	ankrLogo,
	cradlesLogo,
	brightlyStakeLogo,
	midaSwapLogo,
	aspectaLogo,
	echolinkLogo,
	publicAiLogo,
	operatorLogo,
	sideLogo,
	injectiveLogo,
	mellowLogo,
	FlashBotsLogo,
	gearBoxLogo,
	decimalLogo,
	zkbobLogo,
	p2pLogo,
	t2t2Logo,
	hashkeyCloudLogo,
	nodeasyLogo,
	chorusOneLogo,
	availLogo,
	stafiLogo,
	sjtUniversityLogo,
	goSleepLogo,
	infstones,
	staking4allLogo,
	degoLogo,
	promLogo,
	calendarIcon,
	checkmarkCircleIcon,
	clock,
	loadingIcon,
	yellowInfo,
	redAlert,
	greenTick,
	deployIcon,
	backendServer,
	chains,
	frontends,
	instances,
	llms,
	regions,
	arrowUpRight,
	dnsLogo,
	bonfidaLogo,
	spaceIdLogo,
	exploreIcon,
	copyIcon,
	infoIconPurple
};
