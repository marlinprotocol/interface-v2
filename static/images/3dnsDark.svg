<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 20010904//EN"
 "http://www.w3.org/TR/2001/REC-SVG-20010904/DTD/svg10.dtd">
<svg version="1.0" xmlns="http://www.w3.org/2000/svg"
 width="1235.000000pt" height="432.000000pt" viewBox="0 0 1235.000000 432.000000"
 preserveAspectRatio="xMidYMid meet">
<metadata>
Created by potrace 1.16, written by Peter Selinger 2001-2019
</metadata>
<g transform="translate(0.000000,432.000000) scale(0.100000,-0.100000)"
fill="#FFFFFF" stroke="none">
<path d="M0 3350 l0 -340 810 0 c446 0 810 -2 810 -6 0 -4 -261 -445 -362
-611 l-20 -33 470 0 470 0 186 325 c102 178 186 327 186 330 0 3 -202 5 -450
5 -247 0 -450 2 -450 5 0 3 91 148 201 323 111 174 204 323 206 330 4 9 -205
12 -1026 12 l-1031 0 0 -340z"/>
<path d="M10470 3309 c-432 -38 -720 -317 -720 -699 0 -212 86 -398 233 -507
132 -97 222 -130 547 -202 261 -58 272 -62 325 -89 113 -58 168 -189 132 -313
-41 -139 -172 -221 -376 -235 -161 -12 -304 27 -385 104 -47 45 -86 124 -86
175 l0 37 -210 0 -210 0 0 -42 c0 -77 29 -181 72 -267 77 -154 216 -268 403
-330 126 -42 209 -54 370 -54 288 1 503 78 661 238 124 126 186 283 187 475 1
195 -48 328 -162 440 -121 119 -234 166 -596 245 -143 31 -285 68 -315 81
-110 50 -161 124 -168 245 -4 66 -1 84 20 131 63 136 208 205 413 196 81 -4
111 -10 161 -32 103 -46 171 -134 189 -243 l7 -43 210 0 209 0 -6 58 c-21 183
-75 303 -190 418 -167 168 -408 240 -715 213z"/>
<path d="M3490 3085 l0 -185 480 -2 481 -3 -290 -265 -290 -265 0 -142 -1
-141 223 -4 c202 -4 227 -6 279 -27 180 -71 268 -211 255 -406 -12 -189 -108
-314 -280 -367 -103 -31 -239 -23 -330 19 -71 34 -137 95 -168 155 -25 49 -49
142 -49 190 l0 38 -206 0 -207 0 6 -92 c14 -214 104 -397 258 -525 64 -53 210
-126 294 -147 265 -67 558 -26 766 109 124 80 199 163 264 293 56 113 76 206
77 352 0 217 -57 370 -190 511 -93 100 -199 166 -333 210 l-100 33 33 31 c18
18 144 134 281 259 l247 228 0 164 0 164 -750 0 -750 0 0 -185z"/>
<path d="M5310 2099 l0 -1170 533 3 532 4 96 26 c432 118 710 414 806 858 14
68 18 127 17 290 -1 196 -2 210 -32 323 -101 385 -350 653 -720 777 -164 55
-229 60 -754 60 l-478 0 0 -1171z m991 768 c271 -62 456 -241 531 -513 18 -68
22 -105 22 -259 0 -205 -14 -274 -81 -412 -53 -109 -177 -235 -283 -287 -141
-68 -202 -78 -497 -84 l-263 -4 0 792 0 792 253 -5 c172 -3 273 -10 318 -20z"/>
<path d="M7580 2100 l0 -1170 205 0 205 0 0 822 0 822 546 -822 546 -822 209
0 209 0 0 1170 0 1170 -210 0 -210 0 -2 -815 -3 -814 -539 814 -539 815 -209
0 -208 0 0 -1170z"/>
<path d="M270 2152 c0 -17 885 -1515 895 -1515 6 0 106 169 223 375 l213 376
-234 383 -233 384 -432 3 c-238 1 -432 -2 -432 -6z"/>
<path d="M11770 1384 c-45 -19 -106 -78 -133 -129 -16 -30 -21 -58 -21 -110 0
-80 20 -129 76 -182 127 -122 337 -86 421 72 30 56 30 164 0 221 -28 53 -78
102 -128 125 -50 23 -165 25 -215 3z"/>
</g>
</svg>
