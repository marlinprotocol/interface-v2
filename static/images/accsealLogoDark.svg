<svg width="172" height="70" viewBox="0 0 172 70" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M20.6662 22.9934C17.3244 22.9934 15.3563 26.7446 17.2451 29.5052L29.0668 46.7557C29.1725 46.9142 29.3442 47.0066 29.5291 47.0066C32.8709 47.0066 34.8522 43.2554 32.9501 40.4948L21.1417 23.2444C21.036 23.0859 20.8643 22.9934 20.6662 22.9934Z" fill="#288DFF"/>
<path d="M8.10494 35.872L11.6977 41.1158C14.2205 44.7878 18.3944 46.9937 22.8457 46.9937C23.2948 46.9937 23.5722 46.4917 23.308 46.1087L15.8584 35.238C15.7527 35.0795 15.581 34.9871 15.3961 34.9871H8.56725C8.10494 34.9871 7.84077 35.5022 8.10494 35.872Z" fill="#0055FF"/>
<path d="M42.1038 34.115L38.511 28.8712C35.9882 25.186 31.8143 22.9934 27.363 22.9934C26.9139 22.9934 26.6365 23.4953 26.9007 23.8784L34.3503 34.749C34.456 34.9076 34.6277 35 34.8126 35H41.6415C42.0905 34.9868 42.3547 34.4849 42.1038 34.115Z" fill="#38CCFF"/>
<path d="M104.996 38.4346H108.073C108.364 38.4346 108.641 38.6063 108.72 38.8969C108.971 39.8875 109.724 40.363 110.741 40.363C111.943 40.363 112.709 39.7554 112.709 38.778C112.709 35.6476 104.269 37.3383 104.309 31.3416C104.309 28.1451 106.937 26.3619 110.464 26.3619C113.832 26.3619 116.183 27.8809 116.698 30.6019C116.778 31.0114 116.474 31.4076 116.051 31.4076H112.894C112.617 31.4076 112.353 31.2359 112.274 30.9585C112.049 30.1396 111.349 29.6641 110.398 29.6509C109.407 29.6245 108.668 30.1264 108.668 31.1963C108.668 34.1286 117.015 32.8077 117.015 38.461C117.015 41.3008 114.796 43.6652 110.86 43.6652C107.518 43.6652 104.863 42.1462 104.348 39.2535C104.269 38.8176 104.573 38.4346 104.996 38.4346Z" fill="#FFFFFF"/>
<path d="M56.6393 37.2846H59.4791L57.3922 31.103L55.3184 37.2846L54.2353 40.4547L53.3768 42.9907C53.2843 43.2549 53.0333 43.4398 52.756 43.4398H49.8633C49.401 43.4398 49.084 42.9907 49.2425 42.5548L54.8825 26.9819C54.975 26.7177 55.226 26.546 55.5033 26.546H59.347C59.6244 26.546 59.8754 26.7177 59.9679 26.9819L65.6079 42.5548C65.7664 42.9907 65.4494 43.4398 64.9871 43.4398H62.0416C61.7642 43.4398 61.5133 43.2549 61.4208 42.9907L60.5622 40.4547H55.5826L56.6393 37.2846Z" fill="#FFFFFF"/>
<path d="M141.808 37.2846H144.648L142.561 31.103L140.487 37.2846L139.404 40.4547L138.546 42.9907C138.453 43.2549 138.202 43.4398 137.925 43.4398H135.032C134.57 43.4398 134.253 42.9907 134.411 42.5548L140.051 26.9819C140.144 26.7177 140.395 26.546 140.672 26.546H144.516C144.793 26.546 145.044 26.7177 145.137 26.9819L150.777 42.5548C150.935 42.9907 150.618 43.4398 150.156 43.4398H147.211C146.933 43.4398 146.682 43.2549 146.59 42.9907L145.731 40.4547H140.752L141.808 37.2846Z" fill="#FFFFFF"/>
<path d="M157.024 40.2708V27.2207C157.024 26.8509 156.734 26.5603 156.364 26.5603H153.564C153.194 26.5603 152.903 26.8509 152.903 27.2207V42.7937C152.903 43.1635 153.194 43.4541 153.564 43.4541H157.024" fill="#FFFFFF"/>
<path d="M158.346 43.4539H162.48C163.008 43.4539 163.325 42.8595 163.022 42.4236L161.754 40.5744C161.635 40.3895 161.423 40.2838 161.212 40.2838H158.359" fill="#FFFFFF"/>
<path d="M124.967 33.2703V29.8492H131.268C131.796 29.8492 132.113 29.2549 131.809 28.819L130.462 26.8509C130.343 26.666 130.132 26.5603 129.92 26.5603H121.52C121.15 26.5603 120.859 26.8509 120.859 27.2207V42.7937C120.859 43.1635 121.15 43.4541 121.52 43.4541L131.281 43.4409C131.809 43.4409 132.126 42.8465 131.822 42.4106L130.475 40.4425C130.356 40.2576 130.145 40.1519 129.934 40.1519H124.98V36.4403V33.2703H124.967Z" fill="#FFFFFF"/>
<path d="M126.288 36.4534H130.541C131.07 36.4534 131.387 35.859 131.083 35.4231L129.815 33.5739C129.696 33.389 129.485 33.2833 129.273 33.2833H126.288V36.4534Z" fill="#FFFFFF"/>
<path d="M78.6441 38.2626C77.5874 39.3193 76.042 39.8741 74.3777 39.5042C72.687 39.1344 71.3001 37.7739 70.917 36.0832C70.2302 33.0716 72.5021 30.3903 75.3947 30.3903C76.7948 30.3903 78.0365 31.0243 78.8818 32.015C79.0007 32.1603 79.1724 32.2527 79.3705 32.2527H82.6463C83.135 32.2527 83.452 31.7376 83.2406 31.2885C81.8537 28.3562 78.895 26.3221 75.4344 26.3088C70.7585 26.2824 66.7563 30.1922 66.6903 34.868C66.6242 39.7288 70.534 43.6913 75.3815 43.6913C77.5874 43.6913 79.6083 42.8592 81.1405 41.5119C81.3914 41.3006 81.4179 40.9175 81.2329 40.6534L79.6479 38.3419C79.4101 37.9984 78.9346 37.9588 78.6441 38.2626Z" fill="#FFFFFF"/>
<path d="M97.3349 38.2626C96.2783 39.3193 94.7328 39.8741 93.0686 39.5042C91.3779 39.1344 89.991 37.7739 89.6079 36.0832C88.9211 33.0716 91.193 30.3903 94.0856 30.3903C95.4857 30.3903 96.7274 31.0243 97.5727 32.015C97.7048 32.1603 97.8765 32.2527 98.0746 32.2527H101.35C101.839 32.2527 102.156 31.7376 101.945 31.2885C100.558 28.3562 97.5991 26.3221 94.1385 26.3088C89.4626 26.2824 85.4604 30.1922 85.3944 34.868C85.3283 39.7288 89.2381 43.6913 94.0856 43.6913C96.2915 43.6913 98.3124 42.8592 99.8446 41.5119C100.096 41.3006 100.122 40.9175 99.937 40.6534L98.352 38.3419C98.1142 37.9984 97.6387 37.9588 97.3349 38.2626Z" fill="#FFFFFF"/>
</svg>
