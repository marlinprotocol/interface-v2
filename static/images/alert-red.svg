<svg width="25" height="24" viewBox="0 0 25 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M5.82171 9.68293C8.23539 5.41199 9.44222 3.27651 11.0983 2.72681C12.0093 2.4244 12.9907 2.4244 13.9017 2.72681C15.5578 3.27651 16.7646 5.41199 19.1783 9.68293C21.592 13.9539 22.7988 16.0893 22.4368 17.8293C22.2376 18.7866 21.7469 19.6549 21.035 20.3097C19.741 21.5 17.3274 21.5 12.5 21.5C7.67265 21.5 5.25897 21.5 3.96496 20.3097C3.25308 19.6549 2.76239 18.7866 2.56322 17.8293C2.20119 16.0893 3.40803 13.9539 5.82171 9.68293Z" stroke="#E00606" stroke-width="1.5"/>
<path opacity="0.4" d="M12.4922 16H12.5012" stroke="#E00606" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
<path opacity="0.4" d="M12.5 13V9" stroke="#E00606" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
