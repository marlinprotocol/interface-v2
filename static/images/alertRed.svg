<svg width="25" height="25" viewBox="0 0 25 25" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M5.82171 9.83234C8.23539 5.5614 9.44222 3.42592 11.0983 2.87622C12.0093 2.57381 12.9907 2.57381 13.9017 2.87622C15.5578 3.42592 16.7646 5.5614 19.1783 9.83234C21.592 14.1033 22.7988 16.2387 22.4368 17.9787C22.2376 18.936 21.7469 19.8043 21.035 20.4591C19.741 21.6494 17.3274 21.6494 12.5 21.6494C7.67265 21.6494 5.25897 21.6494 3.96496 20.4591C3.25308 19.8043 2.76239 18.936 2.56322 17.9787C2.20119 16.2387 3.40803 14.1033 5.82171 9.83234Z" stroke="#E00606" stroke-width="1.5"/>
<path opacity="0.4" d="M12.4922 16.1494H12.5012" stroke="#E00606" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
<path opacity="0.4" d="M12.5 13.1494V9.14941" stroke="#E00606" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
