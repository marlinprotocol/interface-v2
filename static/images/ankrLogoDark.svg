<svg fill="none" height="34" viewBox="0 0 109 34" width="109" xmlns="http://www.w3.org/2000/svg"><clipPath id="a83e6c200bd1"><path d="m0 0h109v34h-109z"></path></clipPath><g clip-path="url(#a83e6c200bd1)" fill="#ffffff"><path d="m25.76 7.94004-9.53-5.06c-1.17-.62-2.57-.62-3.74 0l-9.62 5.07c-1.46.87-2.81 1.91-2.87 4.89996v2.42h3.75v-3.31l10.62-5.65996 10.54 5.65996v3.31h3.75v-2.42c-.06-3.00996-1.43-4.03996-2.89-4.90996z"></path><path d="m24.91 22.81-8.67 4.85v-3.92c3.05-.72 5.05-3.19 5.05-6.51 0-4.02-2.78-6.73-6.92-6.73s-6.92 2.7-6.92 6.73c0 3.32 2 5.8 5.05 6.51v3.92l-8.75-4.85v-3.62h-3.75v2.42c.06 2.99 1.41 4.03 2.87 4.9l9.63 5.07c1.17.62 2.57.61 3.74 0l9.53-5.05c1.46-.88 2.83-1.9 2.9-4.91v-2.42h-3.75v3.62zm-13.98-5.58c0-2.2 1.22-3.41 3.44-3.41s3.44 1.21 3.44 3.41-1.29 3.42-3.44 3.42-3.44-1.28-3.44-3.42z"></path><path d="m94.1699 9.0902h-5.22l-6.71 6.87v-10.26h-4.3v20.79h4.3v-5.2l2.22-2.27 5.84 7.47h4.83l-7.99-10.22z"></path><path d="m108.21 8.63965c-4.13 0-6.32 1.68995-7.14 2.50995v-2.06995h-4.37v17.40995h4.37v-13.97l7.95-.03-.02-3.84995c-.25 0-.51 0-.79 0z"></path><path d="m68.2601 8.62988c-3.3 0-5.38 2.14002-5.53 2.30002v-1.86002h-4.37v17.42002h4.37v-14.36h3.25.49c1.09.04 1.92.21 2.54.54.2.11.39.24.56.39.93.83 1.17 2.23 1.17 4.33v9.1h4.31v-10.4c0-4.31-1.98-7.46002-6.78-7.46002z"></path><path d="m50.9201 11.4601c-.54-.76-1.17-1.35-1.91-1.79998-1.29-.66-2.75-1.04-4.31-1.04-5.16 0-9.34 4.11998-9.34 9.19998 0 3.83 2.38 7.12 5.77 8.5 1.1.45 2.31.7 3.57.7.8 0 1.57-.1 2.3-.28.02 0 .04-.01.06-.02.08-.02.15-.04.23-.06.32-.1.63-.21.92-.34 1.29-.57 2.2-1.42 2.71-2.39v2.56h4.37v-17.41998h-4.37zm-5.46 11.98c-3.41 0-5.61-2.3-5.61-5.61 0-3.52 2.14-5.61 5.61-5.61 3.06 0 5.61 2.03 5.61 5.61s-2.64 5.61-5.61 5.61z"></path></g></svg>