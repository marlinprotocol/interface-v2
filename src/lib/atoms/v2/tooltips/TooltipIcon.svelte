<script lang="ts">
	import { staticImages } from '$lib/components/images/staticImages';
	import type { TooltipDirection, TooltipVariant } from '$lib/types/componentTypes';

	export let tooltipText: string;
	export let styleClass = '';
	export let iconWidth = '14px';
	export let iconSrc = staticImages.alertV2Icon;
	export let tooltipVariant: TooltipVariant = 'tooltip-secondary';
	export let tooltipDirection: TooltipDirection = 'tooltip-top';

	$: tooltipClass = `${styleClass} ${tooltipVariant} ${tooltipDirection}`;
</script>

<div
	data-testid="tooltip-icon"
	class="tooltip flex shrink-0 text-left font-normal {tooltipClass}"
	data-tip={tooltipText}
>
	<img src={iconSrc} alt="Info" width={iconWidth} />
</div>
