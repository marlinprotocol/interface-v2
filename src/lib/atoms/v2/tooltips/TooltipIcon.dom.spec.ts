import { cleanup, render } from '@testing-library/svelte';
import TooltipIcon from './TooltipIcon.svelte';
import { staticImages } from '$lib/components/images/staticImages';

describe('Tooltip Icon', () => {
	afterEach(() => {
		cleanup();
		vi.resetAllMocks();
	});

	it('tooltipIcon renders with default attribute', () => {
		const { getByTestId } = render(TooltipIcon, { props: { tooltipText: 'Render Tooltip Icon' } });
		const tooltipElement = getByTestId('tooltip-icon');
		const tooltipClasslist = tooltipElement.classList;
		expect(tooltipElement.firstElementChild?.getAttribute('src')).toEqual(staticImages.Alert);
		expect(tooltipElement.firstElementChild?.getAttribute('width')).toEqual('14px');
		expect(tooltipElement.getAttribute('data-tip')).toEqual('Render Tooltip Icon');
		expect(tooltipClasslist.contains('tooltip-secondary')).toBe(true);
		expect(tooltipClasslist.contains('tooltip-top')).toBe(true);
		expect(tooltipElement).toMatchSnapshot();
	});

	it('applies iconSrc attribute to the tooltip Icon', () => {
		const { getByTestId } = render(TooltipIcon, {
			props: {
				tooltipText: 'iconSrc Attribute Test',
				iconSrc: staticImages.CycleLoader
			}
		});
		const tooltipElement = getByTestId('tooltip-icon');
		expect(tooltipElement.firstElementChild?.getAttribute('src')).toEqual(staticImages.CycleLoader);
		expect(tooltipElement.getAttribute('data-tip')).toEqual('iconSrc Attribute Test');
	});

	it('applies tooltipDirection to the tooltipIcon', () => {
		const { getByTestId } = render(TooltipIcon, {
			props: {
				tooltipText: 'tooltipDirection Test',
				tooltipDirection: 'tooltip-bottom'
			}
		});
		const tooltipElement = getByTestId('tooltip-icon');
		const tooltipClasslist = tooltipElement.classList;

		expect(tooltipElement.getAttribute('data-tip')).toEqual('tooltipDirection Test');
		expect(tooltipClasslist.contains('tooltip-bottom')).toBe(true);
	});

	it('applies Icon attribute to the tooltip Icon', () => {
		const { getByTestId } = render(TooltipIcon, {
			props: {
				tooltipText: 'Icon Attribute Test',
				tooltipDirection: 'tooltip-left',
				tooltipVariant: 'tooltip-warning',
				iconSrc: staticImages.Check,
				iconWidth: '20px'
			}
		});
		const tooltipElement = getByTestId('tooltip-icon');
		const tooltipClasslist = tooltipElement.classList;
		expect(tooltipElement.firstElementChild?.getAttribute('src')).toEqual(staticImages.Check);
		expect(tooltipElement.firstElementChild?.getAttribute('width')).toEqual('20px');
		expect(tooltipElement.getAttribute('data-tip')).toEqual('Icon Attribute Test');
		expect(tooltipClasslist.contains('tooltip-left')).toBe(true);
		expect(tooltipClasslist.contains('tooltip-warning')).toBe(true);
	});

	it('applies all attribute to the tooltip Icon', () => {
		const { getByTestId } = render(TooltipIcon, {
			props: {
				tooltipText: 'All Attribute Test',
				tooltipDirection: 'tooltip-left',
				styleClass: 'custom-class',
				tooltipVariant: 'tooltip-warning',
				iconSrc: staticImages.Arrow,
				iconWidth: '20px'
			}
		});
		const tooltipElement = getByTestId('tooltip-icon');
		const tooltipClasslist = tooltipElement.classList;
		expect(tooltipElement.firstElementChild?.getAttribute('src')).toEqual(staticImages.Arrow);
		expect(tooltipElement.firstElementChild?.getAttribute('width')).toEqual('20px');
		expect(tooltipElement.getAttribute('data-tip')).toEqual('All Attribute Test');
		expect(tooltipClasslist.contains('tooltip-left')).toBe(true);
		expect(tooltipClasslist.contains('tooltip-warning')).toBe(true);
		expect(tooltipClasslist.contains('custom-class')).toBe(true);
	});
});
