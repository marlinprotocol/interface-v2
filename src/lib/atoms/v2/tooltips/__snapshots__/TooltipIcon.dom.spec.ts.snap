// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tooltip Icon > tooltipIcon renders with default attribute 1`] = `
<div
  class="tooltip flex shrink-0 text-left font-normal  tooltip-secondary tooltip-top"
  data-testid="tooltip-icon"
  data-tip="Render Tooltip Icon"
>
  <img
    alt="Info"
    src="/images/alert.svg"
    width="14px"
  />
</div>
`;
